src/NLog/Targets/FileTarget.cs

```diff
@@ -996,13 +994,23 @@ private void DateAndSequentialArchive(string fileName, string pattern, LogEventI
 
             int nextSequenceNumber = -1;
 
+            int numDigits = fileTemplate.EndAt - fileTemplate.BeginAt - 2;
             try
             {
                 var directoryInfo = new DirectoryInfo(dirName);
+                int archiveFileNameMinLength = fileNameMask.Length + numDigits;
 #if SILVERLIGHT
-                List<string> files = directoryInfo.EnumerateFiles(fileNameMask).OrderBy(n => n.CreationTime).Select(n => n.FullName).ToList();
+                List<string> files = directoryInfo.EnumerateFiles(fileNameMask)
+                    .Where(n => n.Name.Length >= expectedArchiveFileNameLength)
+                    .OrderBy(n => n.CreationTime)
+                    .Select(n => n.FullName)
+                    .ToList();
 #else
-                List<string> files = directoryInfo.GetFiles(fileNameMask).OrderBy(n => n.CreationTime).Select(n => n.FullName).ToList();
+                List<string> files = directoryInfo.GetFiles(fileNameMask)
+                    .Where(n => n.Name.Length >= archiveFileNameMinLength)
+                    .OrderBy(n => n.CreationTime)
+                    .Select(n => n.FullName)
+                    .ToList();
```

Comment:
We could remove some duplicate code here, by adding an intermediate result  (variable of type `ienumerable<string>`.

the duplicate code:
```
                .OrderBy(n => n.CreationTime)
                   .Select(n => n.FullName)
                    .ToList();

```

could you remove the duplicate code?